receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -156,8 +157,15 @@ func (prw *prometheusRemoteWriteReceiver) handlePRW(w http.ResponseWriter, req *
 		return
 	}
 
+	decompressedBody, err := snappy.Decode(nil, body)
+	if err != nil {
+		prw.settings.Logger.Warn("Error decoding remote write request to snappy", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
```

Comment:
```suggestion
		prw.settings.Logger.Warn("Error decoding snappy-encoded remote write request", zapcore.Field{Key: "error", Type: zapcore.ErrorType, Interface: err})
```

Reply:
Maybe we don't even need this code actually, we just need to understand what is happening in `io.ReadAll(req.Body)` 🤔 

Why is it corrupted over there?

Reply:
Maybe should we decode the snappy before the readall func? Don't know, just a guess